### PR TITLE
Use the standalone command-line tool to destroy runners

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
...without filesystem side effects 🤧 

Closes https://github.com/iterative/cml/issues/1138 along with https://github.com/iterative/terraform-provider-iterative/pull/691 & https://github.com/iterative/terraform-provider-iterative/pull/695.

### CODE REVIEW

**Review by tasdomas — DISMISSED** [review]

File: bin/cml/runner/launch.js [hunk 1]
```diff
@@ -76,6 +76,26 @@ const shutdown = async (opts) => {
     }
   };
 
+  const destroyLeo = async () => {
+    if (!tfResource) return;
+
+    winston.info(`Waiting ${destroyDelay} seconds to destroy`);
+    await sleep(destroyDelay);
+
+    const { cloud, id, name } = JSON.parse(
```
Comment (0x2b3bfa0) [inline bin/cml/runner/launch.js:85]:
```suggestion
    const { cloud, id, name, region } = JSON.parse(
```

File: bin/cml/runner/launch.js [hunk 2]
```diff
@@ -76,6 +76,26 @@ const shutdown = async (opts) => {
...
+      Buffer.from(tfResource, 'base64').toString('utf-8')
+    ).instances[0].attributes;
+
+    try {
+      // FIXME: a crafted name can result in arbitrary code execution
+      return await exec(
+        `leo destroy-runner --cloud="${cloud}" --name="${name}" ${id}`
```
Comment (0x2b3bfa0) [inline bin/cml/runner/launch.js:92]:
```suggestion
        `leo destroy-runner --cloud="${cloud}" --name="${name}" --region="${region}" ${id}`
```

File: bin/cml/runner/launch.js [hunk 3]
```diff
@@ -76,6 +76,26 @@ const shutdown = async (opts) => {
...
+
+    const { cloud, id, name, region } = JSON.parse(
+      Buffer.from(tfResource, 'base64').toString('utf-8')
+    ).instances[0].attributes;
+
+    try {
+      // FIXME: a crafted name can result in arbitrary code execution
+      return await exec(
+        `leo destroy-runner --cloud="${cloud}" --name="${name}" --region="${region}" ${id}`
+      );
```
Comment (0x2b3bfa0) [inline bin/cml/runner/launch.js:93]: * https://github.com/iterative/terraform-provider-iterative/pull/695
```suggestion
      return await exec(
        `leo destroy-runner --cloud=${cloud} --region=${region} ${id}`
      );
```
Still would be safer to escape properly or use [`child_process.execFile`](https://nodejs.org/api/child_process.html#child_processexecfilefile-args-options-callback) but we control these values... more or less. 🙃

File: bin/cml/runner/launch.js [hunk 4]
```diff
@@ -85,7 +105,12 @@ const shutdown = async (opts) => {
     }
   }
 
-  await destroyTerraform();
+  // FIXME: leave LEO only after testing for a while
+  try {
+    await destroyLeo();
+  } catch {
+    await destroyTerraform();
+  }
```
Comment (0x2b3bfa0) [inline bin/cml/runner/launch.js:113]:
```suggestion
  await destroyLeo();
```

**Comment by 0x2b3bfa0** [comment]
Tested using `--cml-version github:iterative/cml#destroy-runner` 🤞🏼

**Review by tasdomas — APPROVED** [review]

**Review by dacbd — APPROVED** [review]
Send it :rocket:

**Pull request merged**